src/fokker_planck.jl

```diff
@@ -0,0 +1,485 @@
+using ..fokker_planck_calculus: assemble_explicit_collision_operator_rhs_serial!
+using ..fokker_planck_calculus: assemble_explicit_collision_operator_rhs_parallel!
+using ..fokker_planck_calculus: assemble_explicit_collision_operator_rhs_parallel_analytical_inputs!
+using ..fokker_planck_calculus: calculate_YY_arrays, enforce_vpavperp_BCs!
+using ..fokker_planck_calculus: calculate_rosenbluth_potential_boundary_data!
+using ..fokker_planck_calculus: enforce_zero_bc!, elliptic_solve!, algebraic_solve!, ravel_c_to_vpavperp_parallel!
+using ..fokker_planck_calculus: calculate_rosenbluth_potentials_via_elliptic_solve!
+using ..fokker_planck_test: Cssp_fully_expanded_form, calculate_collisional_fluxes, H_Maxwellian, dGdvperp_Maxwellian
+using ..fokker_planck_test: d2Gdvpa2_Maxwellian, d2Gdvperpdvpa_Maxwellian, d2Gdvperp2_Maxwellian, dHdvpa_Maxwellian, dHdvperp_Maxwellian
+using ..fokker_planck_test: F_Maxwellian, dFdvpa_Maxwellian, dFdvperp_Maxwellian
+
+########################################################
+# begin functions associated with the weak-form operator
+# where the potentials are computed by elliptic solve
+########################################################
+
+"""
+function that initialises the arrays needed for Fokker Planck collisions
+using numerical integration to compute the Rosenbluth potentials only
+at the boundary and using an elliptic solve to obtain the potentials 
+in the rest of the velocity space domain.
+"""
+
+function init_fokker_planck_collisions_weak_form(vpa,vperp,vpa_spectral,vperp_spectral; precompute_weights=false, test_dense_matrix_construction=false, print_to_screen=true)
+    bwgt = allocate_boundary_integration_weights(vpa,vperp)
+    if vperp.n > 1 && precompute_weights
+        @views init_Rosenbluth_potential_boundary_integration_weights!(bwgt.G0_weights, bwgt.G1_weights, bwgt.H0_weights, bwgt.H1_weights,
+                                        bwgt.H2_weights, bwgt.H3_weights, vpa, vperp, print_to_screen=print_to_screen)
+    end
+    rpbd = allocate_rosenbluth_potential_boundary_data(vpa,vperp)
+    if test_dense_matrix_construction
+        MM2D_sparse, KKpar2D_sparse, KKperp2D_sparse, 
+        KKpar2D_with_BC_terms_sparse, KKperp2D_with_BC_terms_sparse,
+        LP2D_sparse, LV2D_sparse, LB2D_sparse, KPperp2D_sparse,
+        PUperp2D_sparse, PPparPUperp2D_sparse, PPpar2D_sparse,
+        MMparMNperp2D_sparse = assemble_matrix_operators_dirichlet_bc(vpa,vperp,vpa_spectral,vperp_spectral,print_to_screen=print_to_screen)
+    else
+        MM2D_sparse, KKpar2D_sparse, KKperp2D_sparse,
+        KKpar2D_with_BC_terms_sparse, KKperp2D_with_BC_terms_sparse,
+        LP2D_sparse, LV2D_sparse, LB2D_sparse, KPperp2D_sparse,
+        PUperp2D_sparse, PPparPUperp2D_sparse, PPpar2D_sparse,
+        MMparMNperp2D_sparse = assemble_matrix_operators_dirichlet_bc_sparse(vpa,vperp,vpa_spectral,vperp_spectral,print_to_screen=print_to_screen)
+    end

```

Comment:
I never use the dense arrays outside the assemble_matrix_operators_dirichlet_bc* functions. A sparse array is returned by both functions, but the way the sparse array is constructed differs. I would prefer to keep the names as is to make it clear that they ARE sparse arrays, in case someone comes along and makes a new dense matrix that damages the performance.